core/payment-driver/gnt/src/gnt/faucet.rs

```diff
@@ -1,15 +1,37 @@
 use crate::utils;
 use crate::{GNTDriverError, GNTDriverResult};
-use actix_rt::Arbiter;
 use ethereum_types::Address;
 use std::{env, time};
+use trust_dns_resolver::config::{ResolverConfig, ResolverOpts};
+use trust_dns_resolver::TokioAsyncResolver;
 
 const MAX_ETH_FAUCET_REQUESTS: u32 = 6;
 const ETH_FAUCET_SLEEP: time::Duration = time::Duration::from_secs(2);
 const INIT_ETH_SLEEP: time::Duration = time::Duration::from_secs(15);
 const ETH_FAUCET_ADDRESS_ENV_VAR: &str = "ETH_FAUCET_ADDRESS";
 const DEFAULT_ETH_FAUCET_ADDRESS: &str = "http://faucet.testnet.golem.network:4000/donate";
 
+// Hack required on windows to bypass failing trust_dns resolution on windows 10
+// Remove when https://github.com/actix/actix-web/issues/1047 is resolved
+async fn resolve_ip_manually(request_url: String) -> String {
+    let uri: awc::http::Uri = request_url.parse().unwrap();
+    let request_host = uri
+        .host()
+        .expect("Env variable 'ETH_FAUCET_ADDRESS' is not a valid URI.");
```

Comment:
Blowing up the whole process doesn't seem like a proper error handling here. It should return an error.

Reply:
haha, i like to blow stuff up :P

Fixing!

Reply:
resolved